src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ImageListDesigner.cs

```diff
@@ -0,0 +1,355 @@
+    {
+        get
+        {
+            return ImageList.ImageSize;
+        }
+        set
+        {
+            ImageList.Images.Clear();
+            ImageList.ImageSize = value;
+            Images.PopulateHandle();
+        }
+    }
+
+    private bool ShouldSerializeImageSize() => Images.Count == 0;
+
+    private Color TransparentColor
+    {
+        get
+        {
+            return ImageList.TransparentColor;
+        }
+        set
+        {
+            ImageList.Images.Clear();
+            ImageList.TransparentColor = value;
+            Images.PopulateHandle();
+        }
+    }
+
+    private bool ShouldSerializeTransparentColor() => !TransparentColor.Equals(Color.LightGray);
+
+    /// <summary>
+    /// Accessor method for the ImageStream property on ImageList.
+    /// We shadow this property at design time.
+    /// </summary>
+    private ImageListStreamer? ImageStream
+    {
+        get
+        {
+            return ImageList.ImageStream;
+        }
+        set
+        {
+            ImageList.ImageStream = value;
+            Images.ReloadFromImageList();
+        }
+    }
+
+    /// <summary>
+    /// Provides an opportunity for the designer to filter the properties.
+    /// </summary>
+    /// <param name="properties"></param>
+    protected override void PreFilterProperties(IDictionary properties)
+    {
+        base.PreFilterProperties(properties);
+
+        // Handle shadowed properties
+        string[] shadowProps = new string[]
+        {
+            "ColorDepth",
+            "ImageSize",
+            "ImageStream",
+            "TransparentColor"
+        };
+
+        for (int i = 0; i < shadowProps.Length; i++)
+        {
+            PropertyDescriptor? prop = (PropertyDescriptor?)properties[shadowProps[i]];
+            if (prop is not null)
+            {
+                properties[shadowProps[i]] = TypeDescriptor.CreateProperty(typeof(ImageListDesigner), prop, Array.Empty<Attribute>());
+            }
+        }
+
+        // replace this one seperately because it is of a different type (OriginalImageCollection) than
+        // the original property (ImageCollection)
+        PropertyDescriptor? imageProp = (PropertyDescriptor?)properties["Images"];
+        if (imageProp is not null)
+        {
+            Attribute[] attrs = new Attribute[imageProp.Attributes.Count];
+            imageProp.Attributes.CopyTo(attrs, 0);
+            properties["Images"] = TypeDescriptor.CreateProperty(typeof(ImageListDesigner), "Images", typeof(OriginalImageCollection), attrs);
+        }
+    }
+
+    public override DesignerActionListCollection ActionLists
+    {
+        get
+        {
+            _actionLists ??= new DesignerActionListCollection
+            {
+                new ImageListActionList(this)
+            };
+            return _actionLists;
+        }
+    }
+
+    //  Shadow ImageList.Images to allow arbitrary handle recreation.
+    internal class OriginalImageCollection : IList
```

Comment:
You can also add Editor attribute as in original code, it's already ported.
```suggestion
    [Editor($"System.Windows.Forms.Design.ImageCollectionEditor, {AssemblyRef.SystemDesign}", typeof(UITypeEditor))]
    internal class OriginalImageCollection : IList
```